lib/daru/index/index.rb

```diff
@@ -222,6 +222,44 @@ def reorder(new_order)
       self.class.new(new_order.map { |i| from[i] })
     end
 
+    # Sorts a `Index` according to its values. Defaults to ascending order sorting.
+    # raises ArgumentError when all the values are not of the same class.
+    #
+    # == Options
+    #
+    # * +:ascending+ - if false, will sort in descending order. Defaults to true.
+    #
+    # * +:get_new_order+ - if true, then method will return the new order (in list) of
+    # index value along with the sorted `Index`. Defaults to false.
+    #
+    # == Returns:
+    # new sorted `Index` when `get_new_order` is set to false(by default it is false).
+    # sorted `Index` and new order (in a list) of the index value.
+    #
+    # == Usage
+    #
+    #   di = Daru::Index.new [100, 99, 101, 1, 2]
+    #   # Say you want to sort descending order
+    #   di.sort(ascending: false)
+    #   # Want to get the new order as well
+    #   di.sort(get_new_order: true)
+    def sort opts={}
+      raise(ArgumentError, 'All elements are not of the same type/class.') unless check_index_data_class(to_a)
```

Comment:
Why does this line exist? Who said an index has to be the same class for sorting to happen? A symbol and string can be compared. There no need for such a check. Leave it to the user.

Also, why are you passing `to_a` into this method and then calling index_data[0].class`inside the other method? Didn't you think of the overhead that this will have?